gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -52,15 +54,18 @@ public void apply(Project project) {
 
         Provider<Directory> generatedJavaOutputDir = metricSchemaDir.map(file -> file.dir("generated_src"));
         TaskProvider<GenerateMetricSchemaTask> generateMetricsTask = project.getTasks()
-                .register("generateMetrics", GenerateMetricSchemaTask.class, task -> {
+                .register(GENERATE_METRICS, GenerateMetricSchemaTask.class, task -> {
                     task.setGroup(TASK_GROUP);
                     task.setDescription("Generates bindings for producing well defined metrics");
                     task.getInputFile().set(compileSchemaTask.flatMap(CompileMetricSchemaTask::getOutputFile));
                     task.getOutputDir().set(generatedJavaOutputDir);
                 });
-        project.getTasks().named("compileJava", compileJava -> compileJava.dependsOn(generateMetricsTask));
 
-        configureJavaSource(project, generatedJavaOutputDir);
+        SourceSet mainSourceSet = getMainSourceSet(project);
+        mainSourceSet.getJava().srcDirs(generatedJavaOutputDir, generateMetricsTask);
```

Comment:
I'd expect `generateMetricsTask` to be sufficient since it declares outputDir of `generatedJavaOutputDir`, without additional task dependency configuration, but I've been wrong before. @CRogers might have a better idea of how these pieces fit together though

Reply:
Tested and updated -- good call, that does work and generates proper task graph.